torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3049,6 +3137,34 @@ def _finalize_params(fsdp_module: FullyShardedDataParallel) -> None:
                     # reset this flag for cases like "one forward pass + multiple backward passes"
                     self._post_backward_callback_queued = False
 
+        if (self.use_param_exec_order_policy() and self._param_exec_order_prep_stage):
+            self._param_exec_order_prep_stage = False
```

Comment:
IIUC, _finalize_params is only called on FSDP root, so we may have to set it for all child modules as well? Would be good to check this

Reply:
Yes you are right, those are already set here:
https://github.com/pytorch/pytorch/pull/79238/files#diff-ffc1ad1876fbf03d814108f6f2f5950a99281b47e0fd77597745f10dc99add86R3156